src/main/java/at/ac/tuwien/kr/alpha/api/Alpha.java

```diff
@@ -65,47 +77,182 @@ public Alpha(SystemConfig cfg) {
+	public InputProgram readProgramString(String aspString, Map<String, PredicateInterpretation> externals) {
 		ProgramParser parser = new ProgramParser(externals);
 		return parser.parse(aspString);
 	}
 
+	public InputProgram readProgramString(String aspString) {
+		return this.readProgramString(aspString, null);
+	}
+
+	public NormalProgram normalizeProgram(InputProgram program) {
+		return new NormalizeProgramTransformation(this.config.isUseNormalizationGrid()).apply(program);
+	}
+
+	public InternalProgram performProgramPreprocessing(NormalProgram program) {
+		return this.performProgramPreprocessing(InternalProgram.fromNormalProgram(program));
+	}
+
+	public InternalProgram performProgramPreprocessing(InternalProgram program) {
+		LOGGER.debug("Preprocessing InternalProgram!");
+		InternalProgram retVal = program;
+		if (this.config.isEvaluateStratifiedPart()) {
+			AnalyzedProgram analyzed = new AnalyzedProgram(program.getRules(), program.getFacts());
+			retVal = new StratifiedEvaluation().apply(analyzed);
+		}
+		return retVal;
+	}
+
+	public InternalProgram performProgramPreprocessing(AnalyzedProgram program) {
+		LOGGER.debug("Preprocessing AnalyzedProgram!");
+		InternalProgram retVal = program;
+		if (this.config.isEvaluateStratifiedPart()) {
+			retVal = new StratifiedEvaluation().apply(program);
+		}
+		return retVal;
+	}
+
+	/**
+	 * Convenience method - overloaded version of solve({@link InternalProgram}) for cases where details of the program
+	 * analysis and normalization aren't of
+	 * interest
+	 */
+	public Stream<AnswerSet> solve(InputProgram program) {
+		return this.solve(program, InputConfig.DEFAULT_FILTER);
+	}
+
 	/**
-	 * Prepares a solver (and accompanying grounder) instance pre-loaded with the given program. Use this if the solver is needed after reading answer sets
+	 * Convenience method - overloaded version of solve({@link InternalProgram}, {@link Predicate}) for cases where details
+	 * of the program analysis and
+	 * normalization aren't of interest
+	 */
+	public Stream<AnswerSet> solve(InputProgram program, java.util.function.Predicate<Predicate> filter) {
+		NormalProgram normalized = this.normalizeProgram(program);
+		return this.solve(normalized, filter);
+	}
+
+	/**
+	 * Convenience method - overloaded version of solve({@link InternalProgram}) for cases where details of the program
+	 * analysis aren't of interest
+	 */
+	public Stream<AnswerSet> solve(NormalProgram program) {
+		return this.solve(program, InputConfig.DEFAULT_FILTER);
+	}
+
+	/**
+	 * Convenience method - overloaded version of solve({@link InternalProgram}) for cases where details of the program
+	 * analysis aren't of interest
+	 */
+	public Stream<AnswerSet> solve(NormalProgram program, java.util.function.Predicate<Predicate> filter) {
+		InternalProgram preprocessed = this.performProgramPreprocessing(program);
+		return this.solve(preprocessed, filter);
+	}
```

Comment:
Thanks - I removed all unused overloaded methods. Note that I kept the second one since I'm aware of at least one application that depends on Alpha and makes use of that method.